data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/AbstractSqlRepositoryOperations.java

```diff
@@ -523,4 +599,51 @@ public int hashCode() {
     protected interface StatementSupplier<PS> {
         PS create(String ps) throws Exception;
     }
+
+
+    /**
+     * The provider for {@link JsonColumnReader} when JSON columns are being read.
+     *
+     * @param <RS> the result set type
+     */
+    protected static final class JsonColumnReaderProvider<RS> {
```

Comment:
I would have `supports(storedQuery)` and find the first reader that will return true. We might want to have different readers for the same dialect. Plus, we might want to introduce some custom annotation on the method that the reader can detect and provider something specific. One of the scenario might be a different serde reader, now we support multiple implementation like Jackson, Jakarta JSON, BSON etc

Reply:
I have added some customization of that kind, but with SqlPreparedQuery as we are dealing here with Sql JSON column readers. Adding at lower level looked too complex and maybe not needed at the moment.